gatorgrouper/utils/read_student_file.py

```diff
@@ -13,17 +13,30 @@ def read_student_file(filepath):
 
     # read the raw CSV data
     with open(filepath, "r") as csvfile:
-        csvdata = list(csv.reader(csvfile, delimiter=","))
+        has_header = csv.Sniffer().has_header(csvfile.read(1024))
```

Comment:
Is the limit `1024` needed?

Reply:
It runs `has_header` for the first 1024 bytes or whatever the arbitrary number is. If there is no number, it runs it for the entirety of the csv, so it is not completely necessary. The only thing the limitation does is improve efficiency when running the function for csv files.